### DISCUSSION TITLE
flink-cdc MySQL 如何支持从mysql 直接配置数据到kafka？

### DISCUSSION THREAD

**Discussion by chenhaipeng** [apache/flink-cdc#3189] [Q&A]
我想实现从flink-cdc-mysql   到kafka ，但看到官方指引只有到doris ，没有kakfa 的sink 吗？

**Comment by liuxuzxx** [comment]
https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/pipeline-connectors/kafka/  看下这个例子.之前Flink CDC 3.1.x左右的文档好像是没有

**Comment by liuxuzxx** [comment]
但是从kafka如何写入到Doris

就是: mysql--->kafka--->doris  后面这步: kafka--->doris如何做，是否也有yaml配置的形式处理这种方式？

**Reply by WAIYIN-tech** [reply]
+1，请问现在有解决方案了吗😭

**Reply by liuxuzxx** [reply]
有两种解决方案:
第一个方案. 自己编写Sink,写入Doris
第二个方案:使用Doris提供的Routine Load(这个在线下和生产都验证过，对付每天10亿级别的写入量，以及每秒13w/s的速度都是OK的，目前的机器性能只能验证到13w/s的速度，无法提供更高的速度了)，我们目前生产也是使用这个方案。也可以使用Doris提供的Kafka Connector，看到这个操作较为复杂，所以没有选择这个方案

**Reply by liuxuzxx** [reply]
第一个方案. 自己编写Sink,写入Doris   这个方案，之前的同事是这么做的，但是需要自己各种维护代码，尤其是维护人员减少的情况下。不过确实灵活，想怎么定义就怎么定义，想怎么转换就怎么转换。但是转换可以在mysql--->kafka的时候通过Flink自定义函数来实现也可以。后面只要写入就行了

**Reply by liuxuzxx** [reply]
最终我发现，其实性能瓶颈是在Flink CDC同步mysql到Kafka这一步，我们生产的mysql是使用的闪存，Flink CDC按照:8GB的配置，给4个task，大概跑到:6w的速度，还是跟不上 kafka---doris routine load--->doris的速度。所以我只能是首先压入到kafka消息，然后验证下速度
